node/src/chain_spec.rs

```diff
@@ -0,0 +1,491 @@
+                content_config::empty_data_directory_config(),
+                vec![],
+                #[cfg(not(feature = "standalone"))]
+                id,
+            )
+        },
+        vec![],
+        None,
+        None,
+        Some(chain_spec_properties()),
+        #[cfg(not(feature = "standalone"))]
+        Extensions {
+            relay_chain: "rococo-local".into(), // You MUST set this to the correct network!
+            para_id: id.into(),
+        },
+        #[cfg(feature = "standalone")]
+        None,
+    )
+}
+
+pub fn chain_spec_properties() -> json::map::Map<String, json::Value> {
+    let mut properties: json::map::Map<String, json::Value> = json::map::Map::new();
+    properties.insert(
+        String::from("tokenDecimals"),
+        json::Value::Number(json::Number::from(0)),
+    );
+    properties.insert(
+        String::from("tokenSymbol"),
+        json::Value::String(String::from("JOY")),
+    );
+    properties
+}
+
+#[cfg(feature = "standalone")]
+fn session_keys(
+    grandpa: GrandpaId,
+    babe: BabeId,
+    im_online: ImOnlineId,
+    authority_discovery: AuthorityDiscoveryId,
+) -> SessionKeys {
+    SessionKeys {
+        grandpa,
+        babe,
+        im_online,
+        authority_discovery,
+    }
+}
+
+#[allow(clippy::too_many_arguments)]
+pub fn testnet_genesis(
+    #[cfg(feature = "standalone")] initial_authorities: Vec<(
+        AccountId,
+        AccountId,
+        GrandpaId,
+        BabeId,
+        ImOnlineId,
+        AuthorityDiscoveryId,
+    )>,
+    #[cfg(not(feature = "standalone"))] aura_authorities: Vec<AuraId>,
+    root_key: AccountId,
+    endowed_accounts: Vec<AccountId>,
+    cpcp: joystream_node_runtime::ProposalsConfigParameters,
+    members: Vec<membership::genesis::Member<u64, AccountId, Moment>>,
+    forum_config: ForumConfig,
+    data_directory_config: DataDirectoryConfig,
+    initial_balances: Vec<(AccountId, Balance)>,
+    #[cfg(not(feature = "standalone"))] id: ParaId,
+) -> GenesisConfig {
+    #[cfg(feature = "standalone")]
+    const STASH: Balance = 5_000;
+    const ENDOWMENT: Balance = 100_000_000;
+
+    let default_text_constraint = joystream_node_runtime::working_group::default_text_constraint();
+
+    let default_storage_size_constraint =
+        joystream_node_runtime::working_group::default_storage_size_constraint();
+
+    GenesisConfig {
+        frame_system: SystemConfig {
+            code: wasm_binary_unwrap().to_vec(),
+            changes_trie_config: Default::default(),
+        },
+        pallet_balances: BalancesConfig {
+            balances: endowed_accounts
+                .iter()
+                .cloned()
+                .map(|k| (k, ENDOWMENT))
+                // TODO Fix dulicates in initial_authorities
+                // Additional duplicates check in new balances pallet crashes runtime
+                //.chain(initial_authorities.iter().map(|x| (x.0.clone(), STASH)))
```

Comment:
Could you explain here why you are not giving the initial authorities a starting balance?

Reply:
I think its fine when generating the raw chainspec for it to fail if we did indeed have duplicate accounts in initial balances and endowed accounts.

Reply:
Agree, but previously we had the same accounts for both `initial_authorities` and `endowed_accounts` set via `development_config` https://github.com/iorveth/joystream/blob/sumer_parachain/node/src/chain_spec.rs#L144

Made a little filtering to ensure no duplicates were provided. https://github.com/Joystream/joystream/pull/2370/commits/8832b12be3f4a44350bd2f0ee6066b5e4f2c8528